src/systemtask/SystemTask.cpp

```diff
@@ -440,6 +440,7 @@ void SystemTask::UpdateMotion() {
 
   if (stepCounterMustBeReset) {
     motionSensor.ResetStepCounter();
+    motionController.AdvanceDay();
```

Comment:
Can I suggest removing `stepCounterMustBeReset` and the associated message as they aren't needed anymore (I2C is never slept)? This would cause the step count handling to be done immediately when a new day starts. It's a little out of scope for this PR technically but it needs to be refactored anyway and I reckon now is a sensible time to do it

Reply:
It indeed sounds a bit out of scope

I can open a pull request only with this after this one if that's alright

Reply:
No worries, I'll get round to it